strawberry/fastapi/router.py

```diff
@@ -30,10 +31,30 @@ async def __get_root_value():
         return None
 
     @staticmethod
-    async def __get_context(
-        background_tasks: BackgroundTasks, request: Request = None, ws: WebSocket = None
-    ) -> Optional[Any]:
-        return {"request": request or ws, "background_tasks": background_tasks}
+    def __get_context_getter(
+        custom_getter: Callable[..., Optional[Dict[str, Any]]]
+    ) -> Callable[..., Dict[str, Any]]:
+        async def dependency(
+            custom_getter: Dict[str, Any],
+            background_tasks: BackgroundTasks,
+            request: Request = None,
+            ws: WebSocket = None,
+        ) -> Callable[..., Dict[str, Any]]:
```

Comment:
```suggestion
    def __get_context_getter(
        custom_getter: Callable[..., Optional[Dict[str, Any]]]
    ) -> Callable[..., Dict[str, Any]]:
        async def dependency(
            custom_getter: Dict[str, Any],
            background_tasks: BackgroundTasks,
            request: Optional[Request] = None,
            ws: Optional[WebSocket] = None,
        ) -> Callable[..., Dict[str, Any]]:
```

Reply:
Can the `Any`s here be narrowed to `object`s?

Reply:
Do you mean the `Any`s provided to `custom_getter` or just those within the `Callable[]`s?

Reply:
The `Optional`s here trigger typing errors within the fastAPI testing suite so can't be included. Is it an issue if these types are omitted since they have default values of `None`?

Reply:
I don't think you have an option.